presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -83,6 +83,12 @@ public HivePageSourceProvider(
 
     @Override
     public ConnectorPageSource createPageSource(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit split, ConnectorTableHandle table, List<ColumnHandle> columns)
```

Comment:
I think `TestHiveDistributedJoinQueries` should be tested with dynamic filtering on and off

Reply:
Also you could add tests that ensure that with dynamic filtering less input data is read, see `TestWorkProcessorPipelineQueries` for example

Reply:
Sounds good, added `TestHiveDistributedJoinQueriesWithDynamicFiltering`.